synthesis/codon/codon.go

```diff
@@ -85,39 +99,131 @@ type AminoAcid struct {
 
 // Table is an interface that specifies the functions that all table types must implement
 type Table interface {
-	Chooser() (map[string]weightedRand.Chooser, error)
-	GenerateTranslationTable() map[string]string
-	GenerateStartCodonTable() map[string]string
-	GetAminoAcids() []AminoAcid
-	GetStartCodons() []string
-	GetStopCodons() []string
-	IsEmpty() bool
-	OptimizeTable(string) Table
+	GetWeightedAminoAcids() []AminoAcid
```

Comment:
This getter is called in the `fix` package so I kept it, but it probably can be made private by moving whatever `fix` is doing with it to this package here. Weighted Amino acid data only make sense in the context of seq optimisation to me, not sure what a caller would use it for.

Reply:
Maybe make a new issue for that separate to this one?

Reply:
[here you go](https://github.com/TimothyStiles/poly/issues/392)